istio/src/main/scala/io/buoyant/k8s/istio/IstioAttribute.scala

```diff
@@ -0,0 +1,99 @@
+package io.buoyant.k8s.istio
+
+import com.twitter.util.Duration
+
+sealed class MixerAttributeType(name: String)
+
+case object StringAttribute extends MixerAttributeType("string")
+
+case object Int64Attribute extends MixerAttributeType("int64")
+
+case object StringMapAttribute extends MixerAttributeType("map[string, string]")
+
+case object IpAddressAttribute extends MixerAttributeType("ip_address")
+
+case object TimestampAttribute extends MixerAttributeType("timestamp")
+
+case object DurationAttribute extends MixerAttributeType("duration")
+
+/**
+ * From the [https://istio.io/docs/concepts/policy-and-control/attributes.html Istio documentation]
+ *
+ * > Istio uses attributes to control the runtime behavior of services running in the service mesh. Attributes are named and
+ * > typed pieces of metadata describing ingress and egress traffic and the environment this traffic occurs in. An Istio attribute
+ * > carries a specific piece of information such as the error code of an API request, the latency of an API request, or the original
+ * > IP address of a TCP connection
+ *
+ * @param name
+ * @param attributeType
+ * @param description
+ */
+sealed abstract class IstioAttribute[T](val name: String, val attributeType: MixerAttributeType, description: String) {
+  val value: T
+}
+
+case class SourceIpIstioAttribute(val value: String) extends IstioAttribute[String]("source.ip", IpAddressAttribute, "Client IP address.	10.0.0.117")
+
+case class SourceServiceIstioAttribute(val value: String) extends IstioAttribute[String]("source.service", StringAttribute, "The fully qualified name of the service that the client belongs to.	redis-master.my-namespace.svc.cluster.local")
+
+case class SourceNameIstioAttribute(val value: String) extends IstioAttribute[String]("source.name", StringAttribute, "The short name part of the source service.	redis-master")
+
+case class SourceNamespaceIstioAttribute(val value: String) extends IstioAttribute[String]("source.namespace", StringAttribute, "The namespace part of the source service.	my-namespace")
+
+case class SourceDomainIstioAttribute(val value: String) extends IstioAttribute[String]("source.domain", StringAttribute, "The domain suffix part of the source service, excluding the name and the namespace.	svc.cluster.local")
+
+case class SourceUidIstioAttribute(val value: String) extends IstioAttribute[String]("source.uid", StringAttribute, "Platform-specific unique identifier for the client instance of the source service.	kubernetes://redis-master-2353460263-1ecey.my-namespace")
```

Comment:
this might be overly nitpicky, but can we insert some line breaks here? I think lines this long are hard to read especially when my editor wraps them weirdly.

Reply:
Can you let me know what you think of: https://github.com/linkerd/linkerd/pull/1650/files#diff-0a76be1349b81581e237715901125ebcR35 ?

I don't have a particular preference—I usually go with the Editor's default.n Like I mentioned before, it would probably be a good idea if we all shared a single formatter configuration file.

Reply:
I like that alignment style for parameters, but most of the rest of the Linkerd codebase puts parameter lists on the next line, one tab level in, like
```scala
class Foo(
  a: A,
  b: B
) extends Bar {
...
}
```
rather than 
```scala
class Foo(a: A,
          b: B) 
  extends Bar {
...
}
```

Reply:
I agree that we definitely ought to have a shared config file for formatting. Do note that, while we don't have one you can plug into IDEA at the moment, running `sbt test` will also run Scalariform on everything.